### PR TITLE
node_migration: add option to ignore pod health

### PR DESCRIPTION
**Pull request by piax93**
I know there are some efficiency improvements coming up with additional labelling of pods in our pools, but I think it's worth having an option like this to be used in our very large and general purpose pools, in which establishing the health based on pods is a bit of a stretch, and it's quite time/resource consuming to fetch that information out.

### CODE REVIEW

File: clusterman/autoscaler/pool_manager.py [hunk 1]
```diff
@@ -84,12 +84,13 @@ def __init__(
         if fetch_state:
             self.reload_state()
 
-    def reload_state(self) -> None:
+    def reload_state(self, **cluster_connector_kwargs) -> None:
```
Comment (akshaysharma096) [inline clusterman/autoscaler/pool_manager.py:87]: nice.

File: clusterman/kubernetes/kubernetes_cluster_connector.py [hunk 2]
```diff
@@ -94,20 +94,23 @@ def __init__(self, cluster: str, pool: Optional[str], init_crd: bool = False) ->
...
         self._prev_nodes_by_ip = copy.deepcopy(self._nodes_by_ip)
         self._nodes_by_ip = self._get_nodes_by_ip()
-        logger.info("Reloading pods")
-        (self._pods_by_ip, self._unschedulable_pods, self._excluded_pods_by_ip,) = (
-            self._get_pods_info_with_label()
-            if self.pool_config.read_bool("use_labels_for_pods", default=False)
-            else self._get_pods_info()
-        )
+        if load_pods_info:
+            logger.info("Reloading pods")
```
Comment (akshaysharma096) [inline clusterman/kubernetes/kubernetes_cluster_connector.py:106]: I think the log statement should come before the if else clause?

**Review by akshaysharma096 — CHANGES_REQUESTED** [review]

File: clusterman/kubernetes/kubernetes_cluster_connector.py (hunk 2 above)
Comment (piax93) [reply clusterman/kubernetes/kubernetes_cluster_connector.py:106]: I don't think we should write in logs that it's reloading pods, when effectively it's not.

Comment (akshaysharma096) [reply clusterman/kubernetes/kubernetes_cluster_connector.py:106]: A better way is to add a verbose log of what strategy we are using to get the pods `self._pods_by_ip, self._unschedulable_pods, self._excluded_pods_by_ip` 

```py
if load_pods_info:
    logger.info("Reloading pods via label filtering")
else:
   logger.info("Reloading pods without any label filteribg")
```

Comment (piax93) [reply clusterman/kubernetes/kubernetes_cluster_connector.py:106]: But that's outside the scope of this change.

Comment (akshaysharma096) [reply clusterman/kubernetes/kubernetes_cluster_connector.py:106]: Cool!

**Review by akshaysharma096 — APPROVED** [review]
Ship!

**Pull request merged**